unstructured/staging/]

```diff
@@ -0,0 +1,33 @@
+from typing import Any, Dict, List, Optional
+from unstructured.documents.elements import Text
+
+
+def stage_for_datasaur(
+    elements: List[Text],
+    entities: Optional[List[List[Dict[str, Any]]]] = None,
+) -> List[Dict[str, Any]]:
+    """Convert a list of elements into a list of dictionaries for use in Datasaur"""
+    result: List[Dict[str, Any]] = list()
+
+    _entities: List[Dict[str, Any]] = []
+    if entities is not None:
+        if "text" not in entities:
+            raise ValueError("Key 'text' was expected but not present in the Datasaur entities.")
+        _entities = entities
+
+    for item in elements:
+        data = dict(text=item.text, entities=_entities)
+        result.append(data)
+
+    return result
+
+
+def _validate_datasaur_entity(entity: Dict[str, Any]):
+    """Raises an error if the Datasaur entity is invalid."""
+    keys_and_types = {"text": str, "type": str, "start_idx": int, "end_idx": int}
+
+    for key, _type in keys_and_types.items():
+        if key not in entity:
+            raise ValueError(f"Key '{key}' was expected but not present in the Datasaur entity.")
```

Comment:
Pretty sure this file should be deleted.

Reply:
The one named `]`

Reply:
But it's the best file!

Reply:
Even better than `:q!` 🤣